src/createServer.js

```diff
@@ -1,3 +1,35 @@
-// Write code here
-// Also, you can create additional files in the src folder
-// and import (require) them here
+const http = require('http');
+const { validateError } = require('./validateError');
+const { convertToCase } = require('./convertToCase/convertToCase');
+
+const createServer = () => {
+  return http.createServer((req, res) => {
+    const normilizedURL = new URL(req.url, `http://${req.headers.host}`);
+    const originalText = normilizedURL.pathname.slice(1);
+    const targetCase = normilizedURL.searchParams.get('toCase');
+
+    res.setHeader('Content-Type', 'application/json');
+
+    const errors = validateError(originalText, targetCase);
+
+    if (errors.length !== 0) {
+      res.statusCode = 400;
+      res.statusMessage = 'Bad request';
+      res.end(JSON.stringify({ errors }));
+
+      return;
+    }
+
+    const convertedText = convertToCase(originalText, targetCase);
```

Comment:
When I see a text in the naming I expect that it's just a string, but then you spread the string into a new object... If it's an object then don't name it like text.